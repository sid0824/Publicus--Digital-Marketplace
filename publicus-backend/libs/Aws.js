import AWS from 'aws-sdk';
import {AccessKeyId, SecretAccessKey} from "../config/ConfigValues";

export const s3 = new AWS.S3({
  region: 'ap-southeast-2',
  accessKeyId: AccessKeyId,
  secretAccessKey: SecretAccessKey
});

export const ses = new AWS.SES({
  region: 'us-west-2'
});