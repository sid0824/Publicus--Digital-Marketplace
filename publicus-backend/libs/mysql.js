import mysql from 'mysql';
import {MySqlHost, MySqlPassword, MySqlUser} from "../config/ConfigValues";

//mysql connection to database
const connection = mysql.createConnection({
    host: MySqlHost,
    user: MySqlUser,
    password: MySqlPassword,
    database: "survey",
    multipleStatements: true
});

export const runQuery = async (query) => {
    return new Promise((resolve, reject) => {
        connection.query(query, (error, results) => {
            if(error) {
                reject(error);
            } else {
                resolve(results);
            }
        });
    });
};