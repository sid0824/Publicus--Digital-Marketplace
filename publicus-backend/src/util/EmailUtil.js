import { ses } from '../../libs/Aws';
import { SiteUrl, SourceEmail } from "../../config/ConfigValues";

const sendEmail = async (email, subject, body) => {
  const params = {
    Destination: {
      ToAddresses: [
        email
      ]
    },
    Message: {
      Body: {
        Text: {
          Charset: "UTF-8",
          Data: body
        }
      },
      Subject: {
        Charset: "UTF-8",
        Data: subject
      }
    },
    Source: SourceEmail,
  };
  try {
    await ses.sendEmail(params).promise();
  } catch (err) {
    console.error(err);
  }
};

export const sendPurchasedEmail = async (email) => {
  const body = `One of your forms was purchased on the Publicus Marketplace. View in the site.\n${SiteUrl}/myForms`;
  const subject  = "A User Purchased Your Form";
  await sendEmail(email, subject, body);
};

export const sendAcceptedEmail = async (email, form) => {
  const body = `Your form '${form.name}' was approved for the Publicus Marketplace by an administrator.\nIt can now be purchased any used by other users.\n${SiteUrl}/marketplace?search=${form.userId}`;
  const subject  = "Your Form was Approved For the Marketplace!";
  await sendEmail(email, subject, body);
};