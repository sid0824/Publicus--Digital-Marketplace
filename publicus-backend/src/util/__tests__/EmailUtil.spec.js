import { ses } from '../../../libs/Aws';
import {sendAcceptedEmail, sendPurchasedEmail} from "../EmailUtil";

jest.mock('../../../libs/Aws');

describe('EmailUtil', () => {

  test(`should send a purchased email`, async () => {
    ses.sendEmail.mockImplementation((params) => {
      expect(params).toMatchSnapshot();
      return {
        promise: Promise.resolve()
      }
    });
    await sendPurchasedEmail('www.mockEmail.com')
  });

  test(`should send an approval email`, async () => {
    ses.sendEmail.mockImplementation((params) => {
      expect(params).toMatchSnapshot();
      return {
        promise: Promise.resolve()
      }
    });
    await sendAcceptedEmail('www.mockEmail.com', { userId: 'USER_ID', name: 'NAME' })
  });

});