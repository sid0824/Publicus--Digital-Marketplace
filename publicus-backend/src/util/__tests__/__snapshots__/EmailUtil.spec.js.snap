// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailUtil should send a purchased email 1`] = `
Object {
  "Destination": Object {
    "ToAddresses": Array [
      "www.mockEmail.com",
    ],
  },
  "Message": Object {
    "Body": Object {
      "Text": Object {
        "Charset": "UTF-8",
        "Data": "One of your forms was purchased on the Publicus Marketplace. View in the site.
http://publicus-ui.s3-website-ap-southeast-2.amazonaws.com/myForms",
      },
    },
    "Subject": Object {
      "Charset": "UTF-8",
      "Data": "A User Purchased Your Form",
    },
  },
  "Source": "publicusnoreply@gmail.com",
}
`;

exports[`EmailUtil should send an approval email 1`] = `
Object {
  "Destination": Object {
    "ToAddresses": Array [
      "www.mockEmail.com",
    ],
  },
  "Message": Object {
    "Body": Object {
      "Text": Object {
        "Charset": "UTF-8",
        "Data": "Your form 'NAME' was approved for the Publicus Marketplace by an administrator.
It can now be purchased any used by other users.
http://publicus-ui.s3-website-ap-southeast-2.amazonaws.com/marketplace?search=USER_ID",
      },
    },
    "Subject": Object {
      "Charset": "UTF-8",
      "Data": "Your Form was Approved For the Marketplace!",
    },
  },
  "Source": "publicusnoreply@gmail.com",
}
`;
